### PR TITLE
All the fixes! 

### PR DESCRIPTION
**Pull request by amy-hashi**
😮 All the fixes!

- All the configs are now piped to all cluster nodes (not just primaries)
- Added a work-around for a bug with instance groups not being updated when re-creating an instance with the same name
- Correctly boxed OS-specific configs and put them into the right place in the install script
- Lots of little fixes for RHEL on GCP
- Cleaned up everything
- Added example for production mode on RHEL
- Updated the READMEs

### CODE REVIEW

File: primary.tf [hunk 1]
```diff
@@ -1,9 +1,17 @@
...
+  special = false
+  upper   = false
+}
+
 resource "google_compute_instance" "primary" {
   # The number of primaries must be hard coded to 3 when Internal Production Mode
-  # is selected. Currently, that mode does not support scaling. In other modes, the
-  count = "${var.install_type == "ipm" ? 3 : var.primary_count}"
+  # is selected. Currently, that mode does not support scaling.
+  count = "${var.postgresql_user == "none" ? 3 : var.primary_count}"
```
Comment (evanphx) [inline primary.tf]: We're hardcoding this to 3 in AWS, probably do so here too.

File: variables.tf [hunk 2]
```diff
@@ -67,16 +67,44 @@ variable "encryption_password" {
...
+  type        = "string"
+  description = "Preferred zone"
+  default     = "us-central1-a"
+}
+
+###################################################
+# Optional External Services Variables
+###################################################
+
+variable "external_services" {
```
Comment (evanphx) [inline variables.tf]: In AWS, we're detecting if they pass in the postgresql address to decide if we're doing external services implicitly. Do we want to do that here?

Comment (amy-hashi) [reply variables.tf]: AWS is the default value for this, which is why you can skip it. Azure and GCP must pass in the type of external storage.

File: primary.tf (hunk 1 above)
Comment (amy-hashi) [reply primary.tf]: Updated this to be hardcoded to 3.

File: variables.tf (hunk 2 above)
Comment (amy-hashi) [reply variables.tf]: Turns out I wasn't even using this in the code. Removed it from the variables file.

**Pull request merged**